src/Api/Validator.php

```diff
@@ -257,4 +265,19 @@ private function addError($message)
             . ' '
             . $message;
     }
+
+    private function json_encode_type_check($data)
```

Comment:
As discussed over email, "object" can also be encoded regardless of if it is JsonSerializable. So this method could really just be simplified into `!is_resource`. JSON encoding a callable just always generates "{}", so I'd be fine with allowing it.

```php
private function canJsonEncode($data)
{
    return !is_resource($data);
}
```